web/libs/datamanager/src/components/Filters/FilterLine/FilterOperation.jsx

```diff
@@ -56,7 +56,16 @@ export const FilterOperation = observer(({ filter, field, operator, value }) =>
   };
   const availableOperators = filter.cellView?.filterOperators;
   const Input = selected?.input;
-  const operatorList = allowedFilterOperations(types, getRoot(filter)?.SDK?.type);
+  let operatorList = allowedFilterOperations(types, getRoot(filter)?.SDK?.type);
+  if (filter.filter.field.isAnnotationResultsFilter) {
+    // We want at most one of "equal" or "contains" per filter type
+    // They resolve to the same backend query in this custom case
+    operatorList = operatorList.filter((op) => {
+      const hasEqualOperators = operatorList.some((o) => ["equal", "not_equal"].includes(o.key));
+      const allowedOperators = hasEqualOperators ? ["equal", "not_equal"] : ["contains", "not_contains"];
+      return allowedOperators.includes(op.key);
+    });
```

Comment:
I meant this:

```suggestion
    const hasEqualOperators = operatorList.some((o) => ["equal", "not_equal"].includes(o.key));
    const allowedOperators = hasEqualOperators ? ["equal", "not_equal"] : ["contains", "not_contains"];
    operatorList = operatorList.filter((op) => allowedOperators.includes(op.key));
```

`allowedOperators` is common for all operators